### ISSUE TITLE
How to make tagged paragraphs

### ISSUE THREAD

**Issue by rockstorm101** [cpuguy83/go-md2man#27]
Typically when listing the options of a program on a manual page they look like this:

![screenshot from 2017-04-15 15-05-47](https://cloud.githubusercontent.com/assets/15727118/25063765/58cfdd0c-21ed-11e7-92d7-dc4c0bd6f49a.png)

Which is achieved using ".TP". What would be the markdown syntax to achieve this?

**Comment by notramo** [comment]
I think, it shouldn't bring new Markdown syntax.
The options should be written in **bold**, with an optional *argument in italic*.
In a new line, the option description should be written. If the option name is shorter than the indentation of the description, it should be displayed in one line. If the option is longer, the description should be displayed under the option name (of course indented). An option description could contain newline, and a blank line would separate paragraphs.
My idea to detect options:
- It is preceded by a newline (or `# OPTIONS` header)
- It starts with a `-`, and consists of only word characters and `-`.
- Maybe it has an *argument* attached to it with a `=` or space:
  **--input**=*file.txt*
  **-i** *file.txt*
- It can be followed by the long version of it (separated with a `,` and a space), (optionally with argument):
  **-s**, **--short**
  **-i** *file.md*, **--input** *file.md*
- There is nothing other in that line. The description comes in the next line.
  


Example:
```markdown
# OPTIONS
**-b**, **--option-name-in-bold**
The description of the first option.

**-s**
This will be displayed in the same line as the option, because the option is shorter than the indentation. *go-md2man* should indent wrapped lines to the description indentation (currently it is indented to the option indentation)

**-o** *argument*, **--option** *argument*
This is an option with *argument*.
```
This generates the following man page:
![current](https://user-images.githubusercontent.com/26059643/34559240-6c4a2c74-f141-11e7-8080-b858674d2d6a.png)

And it should generate this (indented manually with spaces):
![new](https://user-images.githubusercontent.com/26059643/34559289-8fe683f8-f141-11e7-8059-92816b141913.png)

**Comment by alkisg** [comment]
I'm affected by this too; I can't find any way to get go-md2man create the "indentation level 2" that is needed in OPTIONS, so word wrapping now doesn't occur and I have to split lines manually, which looks horrible.

Example:
```
## OPTIONS
`-o`, `--option`=*name*
    This line now is split whenever I press <enter> in the source
    code; it doesn't word-wrap nicely according to the size of
    the terminal. Maybe we could just use tabs to specify the
    indentation level?
```

**Comment by alkisg** [comment]
It seems that blackfriday, pandoc, and ronn, support lists like that by using a ":" at the start of the line.

I think go-md2man was supposed to support that too, but I couldn't get it to work:
https://github.com/cpuguy83/go-md2man/blob/master/md2man/roff_test.go#L197

I managed to create a "blockquote of indentation level 2" by using " > ", but it leaves an extra line after the option name, so it doesn't look pretty either.

**Comment by alkisg** [comment]
":" indeed works, but only after https://github.com/cpuguy83/go-md2man/commit/391147570fb4e908f72412ff974ca2f2e66bd93a.
Previously, I wasn't able to make it work because all Debian and Ubuntu releases do not have this fix yet.
By manually compiling go-md2man in Debian Buster (which has golang 1.11, as golang <= 1.10 won't do), I was able to make OPTION sections work properly.

So I think this issue needs to be closed as "fixed in https://github.com/cpuguy83/go-md2man/commit/391147570fb4e908f72412ff974ca2f2e66bd93a".
